message/message.go

```diff
@@ -4,41 +4,82 @@
 package msg
 
 import (
+	"fmt"
 	"math/big"
 )
 
-//type MessageType string
+type ChainId uint8
+type TransferType string
+type ResourceId [32]byte
 
-//var DepositAssetType MessageType = "substrate_deposit_asset"
-//var CreateDepositProposalType MessageType = "create_deposit_proposal"
-//var VoteDepositProposalType MessageType = "vote_deposit_proposal"
-//var ExecuteDepositType MessageType = "execute_deposit"
+func (r ResourceId) Hex() string {
+	return fmt.Sprintf("%x", r)
+}
 
-type TransferType string
+type Nonce uint64
+
+func (n Nonce) Big() *big.Int {
+	return big.NewInt(int64(n))
+}
 
 var FungibleTransfer TransferType = "FungibleTransfer"
 var NonFungibleTransfer TransferType = "NonFungibleTransfer"
 var GenericTransfer TransferType = "GenericTransfer"
 
 // Message is used as a generic format to communicate between chains
 type Message struct {
-	Source       ChainId       // Source where message was initiated
-	Destination  ChainId       // Destination chain of message
-	Type         TransferType  // type of bridge transfer
-	DepositNonce uint32        // Nonce for the deposit
-	Metadata     []interface{} // data associated with event sequence
+	Source       ChainId      // Source where message was initiated
+	Destination  ChainId      // Destination chain of message
+	Type         TransferType // type of bridge transfer
+	DepositNonce Nonce        // Nonce for the deposit
+	ResourceId   ResourceId
+	Payload      []interface{} // data associated with event sequence
 }
 
-func NewFungibleTransfer(source, dest ChainId, nonce uint32, amount *big.Int, tokenId []byte, recipient []byte) Message {
+func NewFungibleTransfer(source, dest ChainId, nonce Nonce, amount *big.Int, resourceId ResourceId, recipient []byte) Message {
 	return Message{
 		Source:       source,
 		Destination:  dest,
 		Type:         FungibleTransfer,
 		DepositNonce: nonce,
-		Metadata: []interface{}{
+		ResourceId:   resourceId,
+		Payload: []interface{}{
 			amount.Bytes(),
+			recipient,
+		},
+	}
+}
+
+func NewNonFungibleTransfer(source, dest ChainId, nonce Nonce, resourceId ResourceId, tokenId, recipient, metadata []byte) Message {
+	return Message{
+		Source:       source,
+		Destination:  dest,
+		Type:         NonFungibleTransfer,
+		DepositNonce: nonce,
+		Payload: []interface{}{
+			resourceId,
```

Comment:
Shouldn't this be in the main `Message` struct?

Reply:
I want to confirm we don't have a broken test. This wasn't a failure because we dont have full nft support (WIP) and therefore don't have adequate coverage on it?

Reply:
yeup